src/vm/mlinfo.cpp

```diff
@@ -2796,9 +2796,18 @@ MarshalInfo::MarshalInfo(Module* pModule,
                 }
 #endif // FEATURE_COMINTEROP
 
-                if (m_pMT->HasInstantiation())
+                if (m_pMT->HasInstantiation() && (!m_pMT->IsBlittable()
+                    || m_pMT->HasSameTypeDefAs(g_pByReferenceClass)
+                    || m_pMT->HasSameTypeDefAs(g_pNullableClass)
```

Comment:
What is special about nullable to block it? Maybe worth a comment.

Reply:
@jkoritzinsky asked me to here: https://github.com/dotnet/coreclr/pull/23899#issuecomment-538528286

Its to avoid locking us into the default behavior in case we decide to do something special with the type.

Reply:
Added a comment